services/src/main/java/org/keycloak/protocol/oidc/mappers/LightWeightAccessTokenMapper.java

```diff
@@ -0,0 +1,104 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.keycloak.representations.IDToken.SESSION_ID;
+import static org.keycloak.representations.JsonWebToken.AZP;
+import static org.keycloak.representations.JsonWebToken.SUBJECT;
+
+public class LightWeightAccessTokenMapper extends AbstractOIDCProtocolMapper implements OIDCAccessTokenMapper, EnvironmentDependentProviderFactory {
+    public static String LIGHT_WEIGHT_ACCESS_TOKEN_PROVIDER_ID = "oidc-light-weight-access-token-mapper";
+    private static final List<ProviderConfigProperty> configProperties = new ArrayList<>();
+
+    public static final String EXCLUDE_FROM_ACCESS_TOKEN = "exclude.from.accessToken";
+
+    static {
+        ProviderConfigProperty property = new ProviderConfigProperty();
+        property.setName(EXCLUDE_FROM_ACCESS_TOKEN);
+        property.setLabel("Exclude sid and sub and azp from Access Token");
+        property.setType(ProviderConfigProperty.BOOLEAN_TYPE);
+        property.setDefaultValue("true");
+        property.setHelpText("Exclude sub and sid and azp from Access Token and store them in keycloak's cache memory. " +
+                "The excluded information can be obtained at the token introspection endpoint. " +
+                "If you set this to on and use pairwise subject identifier ," +
+                " you have to set Add to access token of pairwise subject identifier to Off.");
+        configProperties.add(property);
+    }
+
+    @Override
+    public String getDisplayCategory() {
+        return TOKEN_MAPPER_CATEGORY;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "Light weight access token";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Delete the information contained by default in the access token. " +
+                "The deleted information can be obtained at the token introspection endpoint.";
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        return configProperties;
+    }
+
+    @Override
+    public String getId() {
+        return LIGHT_WEIGHT_ACCESS_TOKEN_PROVIDER_ID;
+    }
+
+    @Override
+    public AccessToken transformAccessToken(AccessToken token, ProtocolMapperModel mappingModel, KeycloakSession session,
+                                            UserSessionModel userSession, ClientSessionContext clientSessionCtx) {
+        if (excludeFromAccessToken(mappingModel)) {
+            Map<String, String> data = new HashMap<>();
+            UserSessionModel.SessionPersistenceState persistenceState = userSession.getPersistenceState();
+            if (persistenceState == null || persistenceState == UserSessionModel.SessionPersistenceState.PERSISTENT) {
+                data.put(SESSION_ID, token.getSessionState());
+            } else {
+                // persistenceState is TRANSIENT
+                data.put(SUBJECT, token.getSubject());
```

Comment:
In the case of client credential flow, the session becomes TRANSIENT and the sid is set to null in the processing of the tokenEndpoint after the protocol mapper, so I tried not to set it to SingleUseStore here.　Also, since sub is required to get a TRANSIENT session with findValidSession of the introspection endpoint, it is now set to SingleUseStore.